apps/hash-api/src/graphql/resolvers/knowledge/entity/entity.ts

```diff
@@ -489,3 +492,34 @@ export const isEntityPublicResolver: ResolverFn<
     entityId,
   });
 };
+
+export const getEntityAuthorizationRelationshipsResolver: ResolverFn<
+  Promise<EntityAuthorizationRelationship[]>,
+  {},
+  LoggedInGraphQLContext,
+  QueryIsEntityPublicArgs
+> = async (_, { entityId }, { dataSources, authentication }) => {
+  const context = dataSourcesToImpureGraphContext(dataSources);
+
+  const relationships = await getEntityAuthorizationRelationships(
+    context,
+    authentication,
+    { entityId },
+  );
+
+  return relationships.map(({ object, relation, subject }) => ({
+    objectEntityId: object,
+    relation:
+      relation === "direct_editor"
+        ? EntityAuthorizationRelation.Editor
+        : relation === "direct_owner"
+        ? EntityAuthorizationRelation.Owner
+        : EntityAuthorizationRelation.Viewer,
+    subject:
+      subject.type === "accountGroupMembers"
+        ? { accountGroupId: subject.id as AccountGroupId }
+        : subject.type === "account"
+        ? { accountId: subject.id as AccountId }
+        : { public: true },
+  }));
```

Comment:
Still not sure about doing it with `public: true` (looks stranger to me than `type: "public"`, and you can't easily do `subject.public` anyway), nor separately specifying `accountId` and `accountGroupId` (as it makes it more complicated to gather the subject ids elsewhere, and we don't care whether it's a user or an org anywhere)... let's bear this in mind as a potential simplification for future work here, maybe when we add share links, which will have some kind of code associated with the subject we'll need to factor in.

Reply:
I did consider having something like:
```GQL
enum AuthorizationSubjectType {
  Account
  AccountGroup
  Public
}

type AuthorizationSubject {
  subjectId: SubjectId
  type: AuthorizationSubjectType!
}

type EntityAuthorizationRelationship {
  objectEntityId: EntityId!
  relation: EntityAuthorizationRelation!
  subject: AuthorizationSubject!
}
```

But the generated TS types in the FE would not be able to let TS infer that `subjectId` is non-nullable when `subject.type === "Account"`, or `subject.type === "AccountGroup"`. Is this something you'd rather we sacrifice than go with the approach I've taken here?

Reply:
No, let's not worry about it more for now, we can think about it again when we add more of these objects